UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -287,6 +280,7 @@ private void Init()
 			}
 
 			Inventory.ServerAdd(Spawn.ServerPrefab(pinPrefab).GameObject, pinSlot);
+			FiringPin.gunComp = this.GetComponent<Gun>();
```

Comment:
just ``this`` should be enough since ``this`` is the Gun component, it didn't work like that?